test/integration/roles/test_ecs_ecr/tasks/main.yml

```diff
@@ -5,6 +5,10 @@
 - block:
     - name: When creating with check mode
       ecs_ecr: name='{{ ecr_name }}' region='{{ ec2_region }}'
+      environment:
```

Comment:
Use `profile: {{ profile|default(omit) }}`. 

I think people have to realise if they're running AWS tests that they're responsible for choosing which account it gets run against, but we need to provide a sensible mechanism for that, and profile is probably the best mechanism.